### ISSUE TITLE
no such file assets/shell-app-manifest.json error when generating signed apk

### ISSUE THREAD

**Issue by onka13** [expo/expo#1746]
I've a detached react native project with expokit. It's working very well on debug mode. 
But when I want to generate signed apk, i got an error. When I check the assets folder there is a file which name is 'kernel-manifest.json', not shell-app-manifest. For temporary I copied kernel-manifest file then renamed to shell-app-manifest. Then generating apk process done successfully. Is it ok? if not what should I do?

Error:ENOENT: no such file or directory, open '.......\android\app\src\main\assets\shell-app-manifest.json'

Error:Execution failed for task ':app:bundleProdReleaseExpoAssets'.
Process 'command 'cmd'' finished with non-zero exit value 1

app.json

    "android": {
      "package": "com.test",
      "versionCode": 1,
      "publishBundlePath": "android\\app\\src\\main\\assets\\shell-app.bundle",
      "publishManifestPath": "android\\app\\src\\main\\assets\\shell-app-manifest.json"
      }

public class AppConstants {
  public static final String VERSION_NAME = "2.5.0";
  public static String INITIAL_URL = "exp://exp.host/...";
  public static final boolean IS_DETACHED = true;
  public static final String SHELL_APP_SCHEME = "exp....";
  public static final String RELEASE_CHANNEL = "prod";
  public static boolean SHOW_LOADING_VIEW_IN_SHELL_APP = false;
  public static boolean ARE_REMOTE_UPDATES_ENABLED = true;
  public static final List<Constants.EmbeddedResponse> EMBEDDED_RESPONSES;
  public static boolean FCM_ENABLED = false;

Environment:

exp --version => 53.1.0
react-native --version => 
  react-native-cli: 2.0.1
  react-native: 0.55.2

  OS: Windows 10
  Node: 8.5.0
  Yarn: 1.5.1
  npm: 6.0.0
  Watchman: Not Found
  Xcode: N/A
  Android Studio: Not Found

Packages: (wanted => installed)
  expo: ^27.0.1 => 27.0.1
  react: 16.3.1 => 16.3.1
  react-native: https://github.com/expo/react-native/archive/sdk-27.0.0.tar.gz => 0.55.2

**Comment by marcosacaraujo** [comment]
Hi, I have the same problem. Have you found any solutions to this problem?

**Comment by terribleben** [comment]
Have you published your project with `exp publish` ?

**Comment by onka13** [comment]
Hi @terribleben , I've published with 'exp publish'. Do you need more info?
@marcosacaraujo , change 'kernel-manifest.json' to 'shell-app-manifest.json' worked for me temporary.

**Comment by terribleben** [comment]
When you  run `exp publish` with the `publishBundlePath` and `publishManifestPath` keys present in your `app.json`, it should write something to those file paths. Are you sure those files still don't exist even after running `exp publish`? If they still don't exist, what version of `exp` are you using?

**Comment by onka13** [comment]
@terribleben , I've just run 'exp publish' and shell-app files recreated. So everything is ok now. Thanks for support.

**Comment by pcowgill** [comment]
I had this problem at first, and the solution above of running `exp publish` (with the `publishBundlePath` and `publishManifestPath` keys present in the app.json) before generating a signed APK in Android Studio solved the problem for me too

**Comment by onka13** [comment]
@Solitaryo Bundle name in app.json and file names in asset folder in android must match. You can change publish paths.

"android": {
  "package": "com.test",
  "versionCode": 1,
  "publishBundlePath": "android\\app\\src\\main\\assets\\shell-app.bundle",
  "publishManifestPath": "android\\app\\src\\main\\assets\\shell-app-manifest.json"
  }

**Comment by blackberry4311** [comment]
May I know the purpose of shell-app-manifest.json do in case of detach expo? After published, I saw this file updated and had release channel as well("releaseChannel": "prod-v1" in this case).
But I need to point ios build to channel `prod-v1`(via EXShell.plist) to make this app work and have new feature which I have just published to this channel(prod-v1 channel).

Why don't we just publish to some channel via exp publish and this action auto generate or update the release channel in EXShell.plist or AppConstants.java ?

**Comment by KthProg** [comment]
Running `exp publish` after detaching fixed the issue for me. I'm super confused as to why we need an `exp publish` option after detaching. Expo needs to coordinate better with React Native, because this isn't part of their documentation at all, despite React Native using Expo during the bundle part of the release APK generation. I think that this was causing bugs where my release APKs would not get updated, which was driving me insane.

**Comment by adnanmitf09** [comment]
I got the same issue and after hours of wandering around, I was able to fix this:
**Short answer:** 
Run following command in project directory via terminal.

`expo publish `

**Explanation:**
iOS release build use js bundle located at `os/epluribus/Supporting/shell-app.bundle`
Android release build use js bundle located at `android/app/src/main/assets/shell-app.bundle`
These bundles does not update automatically for ejected projects. So you will lead to a white screen because there is either an old bundle or not at all.

In addition to publishing the bundles online, `expo publish` command creates / modifies the following files (the bundles and configuration files describing where to find the bundles online):

```
android/app/src/main/java/host/exp/exponent/generated/AppConstants.java
android/app/src/main/assets/shell-app-manifest.json
android/app/src/main/assets/shell-app.bundle
ios/react-native-scaffold/Supporting/shell-app-manifest.json
ios/react-native-scaffold/Supporting/shell-app.bundle
```

Thus you would have the same bundle in both debug and release build.
